src/cr/cube/dimension.py

```diff
@@ -1233,6 +1239,30 @@ def _order_dict(self):
         return self._dimension_transforms_dict.get("order") or {}
 
 
+class _SmoothingSpec(object):
+    """Value object providing convenient access to details of a smoothing transform."""
+
+    def __init__(self, smoothing_dict):
+        self._smoothing_dict = smoothing_dict
+
+    @lazyproperty
+    def function(self):
+        """String representing the smoothing function specified in the payload.
+
+        Default smoothing function is ``one_sided_moving_avg``
+        """
+        return self._smoothing_dict.get("function", "one_sided_moving_avg")
+
+    @lazyproperty
+    def window(self):
+        """int, the window parameter specified in the smoothing expression.
+
+        The default value (if window is not specified) is 2 for the moving avarage
+        smoothing algorithms.
+        """
+        return self._smoothing_dict.get("window", 2)
+
```

Comment:
The more I think about this the more I think that a smoothing spec should not exist like this. This works fine for the current case because there is only one type so far. But how would it work if there were two or three or four smoothing types that had different fields?

I think what makes more sense is for `Dimension` to provide the raw smoothing dict from the transforms (keeping in mind the smoothing-dict isn't actually a transform at all, it's just hitching a ride there as an option). This dict is retrieved into a `_BaseSmoother` object over in measures or whatever which feeds it into its factory to produce the right smoother subtype. Then the smoother subtype interprets the dict fields directly. All smoother types have a `"function":` keyword (which serves as its `"type":` field), but all other fields (function arguments) depend on the particular function.

Let me know if you want to discuss to understand this better, but this is the intuition I had initially about this `SmoothingSpec`, it just took me seeing it this third time to actually crystalize the clear rationale for it being elsewhere.

Reply:
I got your point, and it's totally legit. But I think that `_BaseSmoother` should be in the `smoothing` because we smooth measure in both stripe and matrix so if we put this object in the `measure.py` we should duplicate it wheter it's needed in a stripe measure or a matrix measure. I'll try to move the spec somewhere i think it feets. Then we can discuss on this new implementation